beets/importer.py

```diff
@@ -1006,15 +1019,17 @@ def tasks(self):
         for dirs, paths in self.paths():
             if self.session.config['singletons']:
                 for path in paths:
-                    task = self._create(self.singleton(path))
-                    if task:
-                        yield task
+                    tasks = self._create(self.singleton(path))
+                    if tasks:
```

Comment:
To avoid this extra condition (and the one just below), shouldn't `self._create` just return an empty list instead of `None`?